types/src/system/auction/error.rs

```diff
@@ -415,6 +425,21 @@ impl From<bytesrepr::Error> for Error {
     }
 }
 
+impl From<mint::Error> for Error {
+    fn from(error: mint::Error) -> Self {
+        match error {
+            mint::Error::Storage => Self::Storage,
+            mint::Error::MissingKey => Self::MissingKey,
+            mint::Error::CLValue => Self::CLValue,
+            mint::Error::Serialize => Self::Serialization,
+            mint::Error::ArithmeticOverflow => Self::ArithmeticOverflow,
+            mint::Error::GasLimit => Self::GasLimit,
+            mint::Error::InvalidContext => Self::InvalidContext,
+            _ => Self::MintError,
```

Comment:
This _catch-all_ is usually always dangerous. Why do you think it's not here?

Reply:
My suggestion would be to do `.map_err(|_| Error::MintError)` at the call time rather than this wildcard. I think there are just two places you're calling `mint_into_existing_purse.` You could also have two separate variants for each call for better error reporting

Reply:
Applied Michal's suggestion